### PR TITLE
fix createWebhook, executeWebhook(wait=false) and with some code fixes

### PR DESCRIPTION
**Pull request by pchicken**
fixes to two bugs I noticed when using createWebhook and executeWebhook(wait=false)

### CODE REVIEW

**Comment by krisppurg** [comment]
When does message data get nil? @pchicken

**Comment by krisppurg** [comment]
You could append `discard await` when `wait=false`, so `discard await discord.api.executeWebhook(... , wait=false)`.

Unless if that returns nil.

**Comment by krisppurg** [comment]
Ah, I see.
I wouldn't return Message as `Message()` when nil.

The best option I could think of is changing return the `executeWebhook` proc with `Future[Option[Message]]`, same goes any proc that has `wait` in it, since they either produce the same issue;
Example code for my suggestion
```nim
let fut = executeWebhook(... , wait=false)
if fut.isNone:
  echo "There is nothing"

let fut2 = executeWebhook(... , wait=true)
if fut2.isSome:
  echo "Here is the msg: " & $(fut2.get[])
```

File: dimscord/objects.nim [hunk 1]
```diff
@@ -521,7 +521,7 @@ proc newMessage*(data: JsonNode): Message =
         result.author = newUser(data["author"])
     if "member" in data and data["member"].kind != JNull:
         result.member = some newMember(data["member"])
-    if "referenced_message" in data and data["referenced_message"].kind!=JNull:
+    if "referenced_message" in data and data["referenced_message"].kind != JNull:
```
Comment (krisppurg) [inline dimscord/objects.nim:524]: Revert this change, I did `kind!=JNull` on purpose for the sake of not exceeding over 80 characters

File: dimscord/objects.nim [hunk 2]
```diff
@@ -753,7 +753,7 @@ proc newGuildTemplate*(data: JsonNode): GuildTemplate =
         created_at: data["created_at"].str,
         updated_at: data["updated_at"].str,
         source_guild_id: data["source_guild_id"].str,
-        serialized_source_guild:data["serialized_source_guild"].to PartialGuild
+        serialized_source_guild: data["serialized_source_guild"].to PartialGuild
```
Comment (krisppurg) [inline dimscord/objects.nim:756]: Again saving 80 characters

File: dimscord/restapi/message.nim [hunk 3]
```diff
@@ -144,7 +144,7 @@ proc addMessageReaction*(api: RestApi,
         emj = encodeUrl(emoji)
 
     discard await api.request("PUT",
-        endpointReactions(channel_id, message_id, e=emj, uid="@me")
+        endpointReactions(channel_id, message_id, e = emj, uid = "@me")
```
Comment (krisppurg) [inline dimscord/restapi/message.nim:147]: removing the extra spaces is to keep the limit to 80 characters

File: dimscord/restapi/message.nim [hunk 4]
```diff
@@ -157,7 +157,7 @@ proc deleteMessageReaction*(api: RestApi,
 
     discard await api.request(
         "DELETE",
-        endpointReactions(channel_id, message_id, e=emj, uid=user_id)
+        endpointReactions(channel_id, message_id, e = emj, uid = user_id)
```
Comment (krisppurg) [inline dimscord/restapi/message.nim:160]: ^

File: dimscord/restapi/message.nim [hunk 5]
```diff
@@ -174,7 +174,7 @@ proc getMessageReactions*(api: RestApi,
         limit: range[1..100] = 25): Future[seq[User]] {.async.} =
     ## Get all user message reactions on the emoji provided.
     var emj = emoji
-    var url = endpointReactions(channel_id, message_id, e=emj, uid="@me") & "?"
+    var url = endpointReactions(channel_id, message_id, e = emj, uid = "@me") & "?"
```
Comment (krisppurg) [inline dimscord/restapi/message.nim:177]: ^

**Review by krisppurg — COMMENTED** [review]
There are a couple of things I have commented on the following changes; mostly there were about the 80 character limit, since I want to keep the code nicer and polished.

**Comment by krisppurg** [comment]
@pchicken

**Comment by pchicken** [comment]
Sorry

**Comment by krisppurg** [comment]
I'm not exactly asking you to close your PR, you can still open.

**Comment by spx01** [comment]
May I ask why this hasn't been merged yet? The `createWebhook` fix is quite important.

**Comment by krisppurg** [comment]
There are large merge conflicts, and I didnt want to delay dimscord v1.3.0 because of the time it takes. I postponed it, but I'll just close the PR and just add commits based on this PR.

**Comment by krisppurg** [comment]
Due to merge conflicts and issues, I thought it would be a lot more easier if I could just simply make a commit resolving the issue https://github.com/krisppurg/dimscord/commit/8b4dc08f821c6b0f12aab95483765281bc78c359
